src/Microsoft.AspNet.Mvc.TagHelpers/TagHelperOutputExtensions.cs

```diff
@@ -33,21 +33,27 @@ public static class TagHelperOutputExtensions
         {
             if (!tagHelperOutput.Attributes.ContainsName(attributeName))
             {
-                IEnumerable<IReadOnlyTagHelperAttribute> entries;
+                var copiedAttribute = false;
+                for (var i = context.AllAttributes.Count - 1; i >= 0; i--)
+                {
+                    // We look for the original attribute so we can restore the exact attribute name the user typed.
+                    // Approach also ignores changes made to tagHelperOutput[attributeName].
```

Comment:
Update comment to mention restoring the attribute (approximately) where the user wrote it in the Razor source.